### DISCUSSION TITLE
Handle SIGTERM in k8s

### DISCUSSION THREAD

**Discussion by adriangb** [postgresml/pgcat#757] [Q&A] [answered]
It seems like graceful shutdown is only implemented for SIGINT. Kuberentes sends SIGTERM by default to initiate graceful shutdown. Can this be made configurable? AFAIK it is not configurable on the k8s side.

(I also commented on the original PR in https://github.com/postgresml/pgcat/pull/105#discussion_r1648398242 but since it's been closed for a while this may be a better discussion venue).

**Comment by levkk** [comment, marked as answer]
It should be configurable in Docker: https://docs.docker.com/reference/dockerfile/#stopsignal

SIGINT is traditionally for graceful shutdown, SIGTERM is for immediate shutdown.

**Reply by adriangb** [reply]
I don't know what is "traditional", I haven't been around the block that long. It's weird that k8s would choose SIGTERM to initiate graceful shutdown then.

I think k8s does respect the `STOPSIGNAL` set in a Dockerfile (I guess even if not using the Docker engine?), but it seems like pgcat's published image doesn't set it. Is it created from [this Dockerfile](https://github.com/postgresml/pgcat/blob/main/Dockerfile)?

**Reply by adriangb** [reply]
If it is I made https://github.com/postgresml/pgcat/pull/758

**Reply by adriangb** [reply]
I guess the point is that you _want_ `SIGINT` so that PR doesn't make sense from your perspective. I guess I'll just build my own k8s compatible image.

**Reply by levkk** [reply]
I don't have a strong preference for which signal does what; I'm surprised K8s doesn't let you configure the stop signal, it's been an option for container runtimes (e.g. ECS) for a long time. We could make it configurable, but I think it's easier and probably the right thing to do to have your own Dockerfile for your specific deployment. The Dockerfile you linked is the right one, you won't have any trouble building pgcat from source in Docker.

**Reply by adriangb** [reply]
Yeah I've gone and done that.

I'd like to go back on what I said (sorry it's 2am for me): I think #758 should be merged because it doesn't change the current behavior, it just documents it for compatibility with engines that respect that flag (including containerd / whatever GKE uses it seems).
